backend/src/database.ts

```diff
@@ -371,55 +367,57 @@ const pseudoPlayerSchema = new Schema({
   lastGameDate: { type: Date, required: true },
   solvedPuzzles: { type: [String], required: true },
 });
-const PseudoPlayer = mongoose.model("PseudoPlayer", pseudoPlayerSchema);
+const Player = mongoose.model("Player", playerSchema);
 
-// updates the `pseudoPlayer` object (locally) based on the result of a game
-function updatePseudoPlayer(
-  pseudoPlayer: dbPseudoPlayer,
+// updates the `Player` object (locally) based on the result of a game
+function updatePlayer(
+  player: dbPlayer,
   game: GameState,
   score: number,
   newRatingTuple: { rating: number; deviation: number; volatility: number }
 ) {
-  const eloId = pseudoPlayer.eloId;
-  if (eloId !== game.eloIds[0] && eloId !== game.eloIds[1])
+  const idToken = player.idToken;
+
+  if (idToken !== game.idTokens[0] && idToken !== game.idTokens[1])
     console.error("player is not in this game");
-  const pIndex = pseudoPlayer.eloId === game.eloIds[0] ? 0 : 1;
-  pseudoPlayer.name = game.playerNames[pIndex] || "";
-  pseudoPlayer.rating = newRatingTuple.rating;
-  pseudoPlayer.peakRating = Math.max(
-    pseudoPlayer.rating,
-    pseudoPlayer.peakRating
-  );
-  pseudoPlayer.ratingDeviation = newRatingTuple.deviation;
-  pseudoPlayer.ratingVolatility = newRatingTuple.volatility;
-  pseudoPlayer.gameCount++;
-  if (score === 1) pseudoPlayer.winCount++;
-  if (score === 0.5) pseudoPlayer.drawCount++;
-  if (!pseudoPlayer.firstGameDate) pseudoPlayer.firstGameDate = game.startDate;
-  pseudoPlayer.lastGameDate = game.startDate;
+
+  const pIndex = player.idToken === game.idTokens[0] ? 0 : 1;
+  player.name = game.playerNames[pIndex] || "";
+  player.rating = newRatingTuple.rating;
+  player.peakRating = Math.max(player.rating, player.peakRating);
+  player.ratingDeviation = newRatingTuple.deviation;
+  player.ratingVolatility = newRatingTuple.volatility;
+  player.gameCount++;
+  if (score === 1) player.winCount++;
+  if (score === 0.5) player.drawCount++;
+  if (!player.firstGameDate) player.firstGameDate = game.startDate;
+  player.lastGameDate = game.startDate;
 }
 
-// updates both pseudoplayers of a game in the database. If a pseudoplayer is not yet
+// updates both players of a game in the database. If a player is not yet
 // in the database, a new one is created
-async function updatePseudoPlayers(game: GameState): Promise<void> {
-  if (!connectedToDB) return;
-  if (!game.eloIds[0] || !game.eloIds[1]) {
-    console.error(
-      "cannot update pseudo players because game.eloIds is not set"
-    );
+async function updatePlayers(game: GameState): Promise<void> {
+  if (!connectedToDB || isGuest(game.idTokens[0]) || isGuest(game.idTokens[1]))
```

Comment:
I think that even if one player is a guest, the other should still be updated.
We do not want to change its rating (since games with guests are unranked), but we can update other stats like their game count, last game timestamp, and even token (a.k.a. pawn) icon.

Reply:
TODO: add toast notification that explains: "Games with guests will not affect ELO."